### PR TITLE
Megastore: parking lot facelift and fixes

### PR DESCRIPTION
**Pull request by FarkasKBence**
#### Summary
Content "Megastore: new parking lot, fixes"

#### Purpose of change

Partially addresses #62747.

Fixes megastore rotation and gives it a new parking lot.

#### Describe the solution

Rotated megastores 180° so that their entrances face the road.  Also added sidewalk (flags) around megastores.

Redesigned parking lots using Google Maps images of Walmart Supercentres as reference.  The parking layout now should feel more logical, also featuring storage space for trailers and an accesible entrance to the back loading area.

Minor bugs fixes where found.

#### Describe alternatives you've considered

None.

#### Testing

Loaded a world. Previously-loaded megastores and not broken by these changes, and newly generated ones face the right direction.

kép
kép

#### Additional context

Five years I've been doing this and I just find out you can spawn vehicles diagonally.

### CODE REVIEW

File: data/json/obsoletion/obsolete_mapgen.json [hunk 1]
```diff
@@ -501,5 +501,53 @@
     "update_mapgen_id": "nest_RandEnc_VC_Shoppers_remove",
     "//": "Adds a random NPC to the warehouse, placed off in the corner.",
     "object": { "remove_npcs": [ { "class": "RC_survivor_visiting", "x": 0, "y": 0 } ] }
+  },
+  {
+    "type": "mapgen",
```
Comment (Maleclypse) [inline data/json/obsoletion/obsolete_mapgen.json:506]:
```suggestion
    "type": "mapgen"
```

File: data/json/obsoletion/obsolete_mapgen.json [hunk 2]
```diff
@@ -501,5 +501,53 @@
     "update_mapgen_id": "nest_RandEnc_VC_Shoppers_remove",
     "//": "Adds a random NPC to the warehouse, placed off in the corner.",
     "object": { "remove_npcs": [ { "class": "RC_survivor_visiting", "x": 0, "y": 0 } ] }
+  },
+  {
+    "type": "mapgen"
```
Comment (Maleclypse) [inline data/json/obsoletion/obsolete_mapgen.json:506]:
```suggestion
    "type": "mapgen",
```

**Comment by MNG-cataclysm** [comment]
Would you be alright with porting these changes to Defense Mode, please? If not, I can do it in a follow-up PR.

**Comment by LovamKicsiGazsii** [comment]
> Would you be alright with porting these changes to Defense Mode, please? If not, I can do it in a follow-up PR.

I know I'm late on answering, but I agree on that being a follow-up PR.

**Pull request merged**